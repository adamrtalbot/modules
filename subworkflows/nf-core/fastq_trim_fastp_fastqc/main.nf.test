nextflow_workflow {

    name "Test Workflow FASTQ_TRIM_FASTP_FASTQC"
    script "subworkflows/nf-core/fastq_trim_fastp_fastqc/main.nf"
    workflow "FASTQ_TRIM_FASTP_FASTQC"

    test("test_fastq_trim_fastp_fastqc_single_end") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            workflow {
                """
                input_files = [ [ id:'test', single_end:true ], // meta map
                        [ file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true) ]
                    ]
                save_trimmed_fail = false
                save_merged       = false
                skip_fastqc       = false
                skip_fastp        = false

                input[0] = input_files
                input[1] = []
                input[2] = save_trimmed_fail
                input[3] = save_merged
                input[4] = skip_fastqc
                input[5] = skip_fastp
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }


    test("test_fastq_trim_fastp_fastqc_paired_end") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            workflow {
                """
                input_files = [ [ id:'test', single_end:false ], // meta map
                    [ 
                        file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                        file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true) 
                    ]
                ]
                save_trimmed_fail = false
                save_merged       = false
                skip_fastqc       = false
                skip_fastp        = false

                input[0] = input_files
                input[1] = []
                input[2] = save_trimmed_fail
                input[3] = save_merged
                input[4] = skip_fastqc
                input[5] = skip_fastp
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }


    test("test_fastq_trim_fastp_fastqc_interleaved") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            workflow {
                """
                input_files = [ [ id:'test', single_end:true ], // meta map
                        [ file(params.test_data['sarscov2']['illumina']['test_interleaved_fastq_gz'], checkIfExists: true) ]
                    ]
                save_trimmed_fail = false
                save_merged       = false
                skip_fastqc       = false
                skip_fastp        = false

                input[0] = input_files
                input[1] = []
                input[2] = save_trimmed_fail
                input[3] = save_merged
                input[4] = skip_fastqc
                input[5] = skip_fastp
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
