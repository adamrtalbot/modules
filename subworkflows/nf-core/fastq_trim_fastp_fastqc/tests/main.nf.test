nextflow_workflow {

    name "Test Workflow FASTQ_TRIM_FASTP_FASTQC"
    script "subworkflows/nf-core/fastq_trim_fastp_fastqc/main.nf"
    workflow "FASTQ_TRIM_FASTP_FASTQC"

    test("test_fastq_trim_fastp_fastqc_single_end") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            workflow {
                """
                Channel.fromPath(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true)
                    .flatten()
                    .map { it -> tuple( [ id: 'test' ], [ it ] ) }
                    .set { input_files }
                save_trimmed_fail = false
                save_merged       = false
                skip_fastqc       = false
                skip_fastp        = false

                input[0] = input_files
                input[1] = []
                input[2] = save_trimmed_fail
                input[3] = save_merged
                input[4] = skip_fastqc
                input[5] = skip_fastp
                """
            }
        }

        then {
            assert workflow.success
            assert workflow.trace.succeeded().size() == 3
            assert snapshot(
                workflow.out.reads, 
                workflow.out.trim_json, 
                workflow.out.trim_reads_fail, 
                workflow.out.trim_reads_merged, 
                workflow.out.versions
            ).match()
            assert workflow.out.trim_json.size() == 1
            assert workflow.out.trim_html.size() == 1
            assert workflow.out.trim_log.size() == 1
            assert workflow.out.fastqc_raw_html.size() == 0
            assert workflow.out.fastqc_raw_zip.size() == 0
            assert workflow.out.fastqc_trim_html.size() == 1
            assert workflow.out.fastqc_trim_zip.size() == 1
        }

    }


    test("test_fastq_trim_fastp_fastqc_paired_end") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            workflow {
                """
                in_1 = Channel.fromPath(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true)
                in_2 = Channel.fromPath(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true)
                in_1.mix(in_2)
                    .collect()
                    .map { it -> tuple( [ id: 'test' ], it ) }
                    .set { input_files }
                save_trimmed_fail = false
                save_merged       = false
                skip_fastqc       = false
                skip_fastp        = false

                input[0] = input_files
                input[1] = []
                input[2] = save_trimmed_fail
                input[3] = save_merged
                input[4] = skip_fastqc
                input[5] = skip_fastp
                """
            }
        }

        then {
            assert workflow.success
            assert workflow.trace.succeeded().size() == 3
            assert snapshot(
                workflow.out.reads, 
                workflow.out.trim_json, 
                workflow.out.trim_reads_fail, 
                workflow.out.trim_reads_merged, 
                workflow.out.versions
            ).match()
            assert workflow.out.trim_json.size() == 1
            assert workflow.out.trim_html.size() == 1
            assert workflow.out.trim_log.size() == 1
            assert workflow.out.fastqc_raw_html.size() == 0
            assert workflow.out.fastqc_raw_zip.size() == 0
            assert workflow.out.fastqc_trim_html.size() == 1
            assert workflow.out.fastqc_trim_zip.size() == 1
        }

    }


    test("test_fastq_trim_fastp_fastqc_interleaved") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            workflow {
                """
                Channel.fromPath(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true)
                    .flatten()
                    .map { it -> tuple( [ id: 'test' ], it ) }
                    .set { input_files }
                save_trimmed_fail = false
                save_merged       = false
                skip_fastqc       = false
                skip_fastp        = false

                input[0] = input_files
                input[1] = []
                input[2] = save_trimmed_fail
                input[3] = save_merged
                input[4] = skip_fastqc
                input[5] = skip_fastp
                """
            }
        }

        then {
            assert workflow.success
            assert workflow.trace.succeeded().size() == 3
            assert snapshot(
                workflow.out.reads, 
                workflow.out.trim_json, 
                workflow.out.trim_reads_fail, 
                workflow.out.trim_reads_merged, 
                workflow.out.versions
            ).match()
            assert workflow.out.trim_json.size() == 1
            assert workflow.out.trim_html.size() == 1
            assert workflow.out.trim_log.size() == 1
            assert workflow.out.fastqc_raw_html.size() == 0
            assert workflow.out.fastqc_raw_zip.size() == 0
            assert workflow.out.fastqc_trim_html.size() == 1
            assert workflow.out.fastqc_trim_zip.size() == 1
        }

    }

}
