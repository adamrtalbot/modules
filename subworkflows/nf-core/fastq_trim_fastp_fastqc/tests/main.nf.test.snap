{
    "test_fastq_trim_fastp_fastqc_single_end": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.fastp.fastq.gz:md5,dc08e42319d0deab088da4d8600aa2bd"
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test.fastp.json:md5,c852d7a6dba5819e4ac8d9673bedcacc"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    "versions.yml:md5,54e38ebe354c705a56c06a11c0ccf68e",
                    "versions.yml:md5,712d34b9d3e3703d173d6422327e6f9e",
                    "versions.yml:md5,2da05fb640ca6106be6412bd32e8003f"
                ]
            ]
        ],
        "timestamp": "2023-09-08T13:53:49.356875"
    },
    "test_fastq_trim_fastp_fastqc_paired_end": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    [
                        "test_1.fastp.fastq.gz:md5,dc08e42319d0deab088da4d8600aa2bd",
                        "test_2.fastp.fastq.gz:md5,e59d0d65cc8c2801f1b970df601fc207"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test.fastp.json:md5,1e0f8e27e71728e2b63fc64086be95cd"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    "versions.yml:md5,712d34b9d3e3703d173d6422327e6f9e",
                    "versions.yml:md5,2da05fb640ca6106be6412bd32e8003f",
                    "versions.yml:md5,54e38ebe354c705a56c06a11c0ccf68e"
                ]
            ]
        ],
        "timestamp": "2023-09-08T13:54:11.942043"
    },
    "test_fastq_trim_fastp_fastqc_interleaved": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.fastp.fastq.gz:md5,dc08e42319d0deab088da4d8600aa2bd"
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test.fastp.json:md5,c852d7a6dba5819e4ac8d9673bedcacc"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    "versions.yml:md5,54e38ebe354c705a56c06a11c0ccf68e",
                    "versions.yml:md5,712d34b9d3e3703d173d6422327e6f9e",
                    "versions.yml:md5,2da05fb640ca6106be6412bd32e8003f"
                ]
            ]
        ],
        "timestamp": "2023-09-08T13:54:32.228126"
    }
}